### DISCUSSION TITLE
How to use the open-meteo historical forecast data for bias correction

### DISCUSSION THREAD

**Discussion by TaZeppe** [open-meteo/open-meteo#1023] [General]
Hi 
and thanks for the amazing api and easy accessibility of complex weather data. I have been following the project for some time and now want to understand the historical forecast data api better and how this data can be utilized for building ML weather models. 
As far as I understand from your great post: https://substack.com/@patrickzippenfenig/p-145599416 the fetch of the last 24 hours of data for e.g. the ICON D2 model will result in the 3 hour forecast for 24/3=8 different model runs, correct? In the case of the ECMWF IFS 0.25° operational model each model run takes approximately 5 hours and 20 minutes to complete and first at hour 6 will you have hours 0-6 for the same day. Thus, if your objective is to e.g. bias correct a specific weather parameter for the IFS operational forecast you are training on data that will not be available when running the model live. Certainly you can use the forecast that is available at the time you want to make the predictions, but if you want to evaluate your models performance/minimize the error between hour 7 for modelrun 0 (available at hour 6) you need to have a dataset only consisting of this modelrun, which as far as I understand, you cannot select at the moment? Or am I missing something? How would you use the historical forecast data for improving future forecasts? 
thanks in advance.
Talia

**Comment by patrick-zippenfenig** [comment]
Hi,
correct. The Historical Forecast API assembles the first hours of every weather model run to be as close to initialisation (~measurements) data as possible.

To train a ML model with a lead-time offset, you can use the [previous model run API](https://open-meteo.com/en/docs/previous-runs-api). It does not give you access to individual runs (e.g. only 0z runs), but it maintains a lead-time offset of 24, 48 or more hours. The typical use-case is training a model for the day-ahead model. This system might be extended to lead-time offsets of 6 and 12 hours as well.

**Reply by TaZeppe** [reply]
Thanks for the reply, I wrote my follow up comment at the same time you wrote your reply - my question below still stands, but I apologize for not having read that blog post first.

**Comment by TaZeppe** [comment]
Hi again, I have now also read this blog post https://openmeteo.substack.com/p/weather-forecasts-from-previous-model-runs, which partly answers my question above - however, do we agree that if I want to make a model that makes predictions at hour 7 for the 24 hours of the following day using the previous day 1 forecast as input when optimizing, the data used as input will (in the case of the IFS model) be a combination of modelrun 00, 06, 12 and 18? Thus hours 0-6 will be modelrun 00, hours 6-12 will be modelrun 06 and so on - even though modelrun 18 will be the latest forecast for all 24 hours?

**Reply by patrick-zippenfenig** [reply]
Yes, this is correct. Technically the lead-time offset should therefore include the model and processing delay. So a 24 hour lead-time is in reality 24 hours + 5:20 for processing.

Working with customers, this approach worked well for day-ahead forecasting in combination with time-series aware ML models and outperformed only considering 0z.
